tests/unit/test_discovery_client.py

```diff
@@ -76,7 +77,7 @@ def test___discovery_service_available___unregister_non_registered_service___unr
 ):
     unregistration_success_flag = discovery_client.unregister_service()
 
-    assert ~unregistration_success_flag  # False
+    assert unregistration_success_flag
```

Comment:
Is this supposed to fail or not? The test name says failure but the assertion says success.

Reply:
This test case scenario is:
   1. When measurement service is not registered with disc service.
   2. We cannot unregister a service which is not register at all.
   3. So, this logs and info that `Not registered with discovery service.`

And yes assertion can be updated to `unregistration_failure_flag`

Reply:
`test___discovery_service_available___unregister_non_registered_service___unregistration_failure` means "Given that the discovery service is available, when I unregister a non-registered service, then I get an unregistration failure".

True means success and False means failure. The assertion expects True, which means success. However, the test name said you should get failure, not success. Which is it?

It makes no sense to rename the variable to mean the opposite of what it actually means.

Reply:
Okay, now I have updated the `unregister_service` method to return `False` if we are trying to unregister a non-registered service. And in test, I am expecting the assertion to be `False`.